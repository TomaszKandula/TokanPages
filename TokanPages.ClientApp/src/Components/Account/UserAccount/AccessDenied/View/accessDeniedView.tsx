import * as React from "react";
import { Link } from "react-router-dom";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Container from "@material-ui/core/Container";
import Skeleton from "@material-ui/lab/Skeleton";
import { Button, Typography } from "@material-ui/core";
import { ReactHtmlParser } from "../../../../../Shared/Services/Renderers";
import { CustomDivider } from "../../../../../Shared/Components";
import { AccessDeniedProps } from "../accessDenied";

interface AccessDeniedViewProps extends AccessDeniedProps {
    isLoading: boolean;
    languageId: string;
    accessDeniedCaption: string;
    accessDeniedPrompt: string;
    homeButtonText: string;
}

const HomeButton = (props: AccessDeniedViewProps): React.ReactElement => {
    return (
        <Link to={`/${props.languageId}`} className="link" rel="noopener nofollow">
            <Button fullWidth variant="contained" className="button" disabled={props.isLoading}>
                {props.homeButtonText}
            </Button>
        </Link>
    );
};

export const AccessDeniedView = (props: AccessDeniedViewProps): React.ReactElement => {
    return (
        <section className={`section ${props.background ?? ""}`}>
            <Container className="container-wide">
                <div className="pt-120 pb-64">
                    <Card elevation={0} className="card">
                        <CardContent className="card-content">
                            <Typography component="span" className="access-denied-caption">
                                {props.isLoading ? <Skeleton variant="text" /> : props.accessDeniedCaption}
                            </Typography>
                            <CustomDivider mt={2} mb={1} />
                            <div className="pt-25 pb-25">
                                <Typography component="span" className="access-denied-prompt">
                                    {props.isLoading ? (
                                        <Skeleton variant="text" height="100px" />
                                    ) : (
                                        <ReactHtmlParser html={props.accessDeniedPrompt} />
                                    )}
                                </Typography>
                            </div>
                            {props.isLoading ? (
                                <Skeleton variant="rect" width="100%" height="40px" />
                            ) : (
                                <HomeButton {...props} />
                            )}
                        </CardContent>
                    </Card>
                </div>
            </Container>
        </section>
    );
};
