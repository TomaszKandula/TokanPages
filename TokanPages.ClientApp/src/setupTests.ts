// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
