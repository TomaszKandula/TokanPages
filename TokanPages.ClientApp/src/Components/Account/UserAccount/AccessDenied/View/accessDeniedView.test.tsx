import "../../../../../setupTests";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { AccessDeniedView } from "./accessDeniedView";

describe("test account group component: accessDeniedView", () => {
    it("should render correctly '<AccessDeniedView />' when content is loaded.", () => {
        const html = render(
            <BrowserRouter>
                <AccessDeniedView
                    isLoading={false}
                    accessDeniedCaption={"Access Denied"}
                    accessDeniedPrompt={"<p>You do not have access to this page.</p><p>If you believe this is an error, please contact IT support.</p>"}
                    homeButtonText={"Go back"}
                />
            </BrowserRouter>
        );

        expect(html).toMatchSnapshot();
    });
});
