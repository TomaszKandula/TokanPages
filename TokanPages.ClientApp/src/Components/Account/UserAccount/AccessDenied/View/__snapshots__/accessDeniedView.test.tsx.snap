// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test account group component: accessDeniedView should render correctly '<AccessDeniedView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section>
        <div
          class="bulma-container bulma-is-max-desktop"
        >
          <div
            class="py-6"
          >
            <div
              class="bulma-card"
            >
              <div
                class="bulma-card-content"
              >
                <div
                  class="custom-card-icon-background is-flex is-justify-content-center my-6"
                >
                  <div
                    class="custom-card-icon-holder is-flex is-justify-content-center is-align-items-center"
                  >
                    <div>
                      <svg
                        class="is-flex is-align-self-center custom-card-icon has-text-danger"
                        role="presentation"
                        style="width: 4.5rem; height: 4.5rem;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12,2C17.53,2 22,6.47 22,12C22,17.53 17.53,22 12,22C6.47,22 2,17.53 2,12C2,6.47 6.47,2 12,2M15.59,7L12,10.59L8.41,7L7,8.41L10.59,12L7,15.59L8.41,17L12,13.41L15.59,17L17,15.59L13.41,12L17,8.41L15.59,7Z"
                          style="fill: currentColor;"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <h2
                  class="bulma-title has-text-centered"
                >
                  Access Denied
                </h2>
                <div
                  class="my-5"
                >
                  <p
                    class="is-size-6 has-text-centered line-height-20"
                  >
                    You do not have access to this page.
                  </p>
                  <p
                    class="is-size-6 has-text-centered line-height-20"
                  >
                    If you believe this is an error, please contact IT support.
                  </p>
                </div>
                <div
                  class="mb-4"
                />
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section>
      <div
        class="bulma-container bulma-is-max-desktop"
      >
        <div
          class="py-6"
        >
          <div
            class="bulma-card"
          >
            <div
              class="bulma-card-content"
            >
              <div
                class="custom-card-icon-background is-flex is-justify-content-center my-6"
              >
                <div
                  class="custom-card-icon-holder is-flex is-justify-content-center is-align-items-center"
                >
                  <div>
                    <svg
                      class="is-flex is-align-self-center custom-card-icon has-text-danger"
                      role="presentation"
                      style="width: 4.5rem; height: 4.5rem;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12,2C17.53,2 22,6.47 22,12C22,17.53 17.53,22 12,22C6.47,22 2,17.53 2,12C2,6.47 6.47,2 12,2M15.59,7L12,10.59L8.41,7L7,8.41L10.59,12L7,15.59L8.41,17L12,13.41L15.59,17L17,15.59L13.41,12L17,8.41L15.59,7Z"
                        style="fill: currentColor;"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <h2
                class="bulma-title has-text-centered"
              >
                Access Denied
              </h2>
              <div
                class="my-5"
              >
                <p
                  class="is-size-6 has-text-centered line-height-20"
                >
                  You do not have access to this page.
                </p>
                <p
                  class="is-size-6 has-text-centered line-height-20"
                >
                  If you believe this is an error, please contact IT support.
                </p>
              </div>
              <div
                class="mb-4"
              />
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
